### ISSUE TITLE
commit 2766faa8 contains an invalid filename

### ISSUE THREAD

**Issue by mdaniel** [VincentToups/emacs-utils#2]
I was trying to clone the repository when msysgit exploded with a very strange error.

As it turns out, the error came from a bogus file in the repository, which landed in commit 2766faa8fe1a8805fe7a74a8bbd76a7e51903afd

The filename is `ls: cannot access page-*.el: No such file or directory`

If you can remove that file, it will permit more successful clones of your repository. :-)

**Comment by VincentToups** [comment]
Matthew,

Thanks for the heads up.  I only ever push to the repo.  I'll send you
another email when its fixed.

-V

On Mon, Oct 10, 2011 at 1:25 PM, Matthew Daniel
reply@reply.github.com
wrote:

**Comment by VincentToups** [comment]
Matthew,

I've pushed a new version of the repository to github after deleting
the offending file.

Incidentally, are you on windows?  I'm able to clone the repository
without any problems using command line get in Ubuntu Natty.

I've never extensively tested these utilities on Windows, but I think
only the scripting.el code would be effected.

-V

**Comment by mdaniel** [comment]
Hi Vincent,

Yes, it was on Windows but that is why I said that you will open up
the repository to a wider audience. :-)

Thank you for looking into this, and for providing your cool project
to the world.

  -- /v\atthew

On Mon, Oct 10, 2011 at 7:33 PM, Vincent Toups
reply@reply.github.com
wrote:

> Matthew,
> 
> I've pushed a new version of the repository to github after deleting
> the offending file.
> 
> Incidentally, are you on windows?  I'm able to clone the repository
> without any problems using command line get in Ubuntu Natty.
> 
> I've never extensively tested these utilities on Windows, but I think
> only the scripting.el code would be effected.
> 
> -V
> 
> On Mon, Oct 10, 2011 at 1:26 PM, J. Vincent Toups

**Comment by VincentToups** [comment]
If you don't mind me asking, what part of my emacs repository are you
interested in?

There is a lot of stuff in there.

-V

On Mon, Oct 10, 2011 at 1:36 PM, Matthew Daniel
reply@reply.github.com
wrote:

> Hi Vincent,
> 
> Yes, it was on Windows but that is why I said that you will open up
> the repository to a wider audience. :-)
> 
> Thank you for looking into this, and for providing your cool project
> to the world.
> 
>  -- /v\atthew
> 
> On Mon, Oct 10, 2011 at 7:33 PM, Vincent Toups
> reply@reply.github.com
> wrote:
> 
> > Matthew,
> > 
> > I've pushed a new version of the repository to github after deleting
> > the offending file.
> > 
> > Incidentally, are you on windows?  I'm able to clone the repository
> > without any problems using command line get in Ubuntu Natty.
> > 
> > I've never extensively tested these utilities on Windows, but I think
> > only the scripting.el code would be effected.
> > 
> > -V
> > 
> > On Mon, Oct 10, 2011 at 1:26 PM, J. Vincent Toups

**Comment by mdaniel** [comment]
This article: http://www.reddit.com/r/emacs/comments/l46cl/i_updated_my_emacs_lisp_monadic_parser_combinator/

led me to your blog post:
http://dorophone.blogspot.com/2011/10/update-to-my-monadic-parser-combinator.html

and I am always interested in trying out new parsing tools. I don't
have any experience in playing with a parsing tool that is hosted in
Emacs, so I wanted to try it out.

I actually didn't examine the repository closely (as I dislike digging
around in repositories via the github interface), but I am always
excited to learn new elisp tricks, too. :-)

  -- /v\atthew

On Mon, Oct 10, 2011 at 7:43 PM, Vincent Toups
reply@reply.github.com
wrote:

> If you don't mind me asking, what part of my emacs repository are you
> interested in?
> 
> There is a lot of stuff in there.
> 
> -V
> 
> On Mon, Oct 10, 2011 at 1:36 PM, Matthew Daniel
> reply@reply.github.com
> wrote:
> 
> > Hi Vincent,
> > 
> > Yes, it was on Windows but that is why I said that you will open up
> > the repository to a wider audience. :-)
> > 
> > Thank you for looking into this, and for providing your cool project
> > to the world.
> > 
> >  -- /v\atthew
> > 
> > On Mon, Oct 10, 2011 at 7:33 PM, Vincent Toups
> > reply@reply.github.com
> > wrote:
> > 
> > > Matthew,
> > > 
> > > I've pushed a new version of the repository to github after deleting
> > > the offending file.
> > > 
> > > Incidentally, are you on windows?  I'm able to clone the repository
> > > without any problems using command line get in Ubuntu Natty.
> > > 
> > > I've never extensively tested these utilities on Windows, but I think
> > > only the scripting.el code would be effected.
> > > 
> > > -V
> > > 
> > > On Mon, Oct 10, 2011 at 1:26 PM, J. Vincent Toups

**Comment by VincentToups** [comment]
Matt,

This post summarizes some of the projects in the repository.  Let me
know if you have any questions about them!

http://dorophone.blogspot.com/2011/04/emacs-lisp-curios-some-useful-some.html

-V

On Mon, Oct 10, 2011 at 2:44 PM, Matthew Daniel
reply@reply.github.com
wrote:

> This article: http://www.reddit.com/r/emacs/comments/l46cl/i_updated_my_emacs_lisp_monadic_parser_combinator/
> 
> led me to your blog post:
> http://dorophone.blogspot.com/2011/10/update-to-my-monadic-parser-combinator.html
> 
> and I am always interested in trying out new parsing tools. I don't
> have any experience in playing with a parsing tool that is hosted in
> Emacs, so I wanted to try it out.
> 
> I actually didn't examine the repository closely (as I dislike digging
> around in repositories via the github interface), but I am always
> excited to learn new elisp tricks, too. :-)
> 
>  -- /v\atthew
> 
> On Mon, Oct 10, 2011 at 7:43 PM, Vincent Toups
> reply@reply.github.com
> wrote:
> 
> > If you don't mind me asking, what part of my emacs repository are you
> > interested in?
> > 
> > There is a lot of stuff in there.
> > 
> > -V
> > 
> > On Mon, Oct 10, 2011 at 1:36 PM, Matthew Daniel
> > reply@reply.github.com
> > wrote:
> > 
> > > Hi Vincent,
> > > 
> > > Yes, it was on Windows but that is why I said that you will open up
> > > the repository to a wider audience. :-)
> > > 
> > > Thank you for looking into this, and for providing your cool project
> > > to the world.
> > > 
> > >  -- /v\atthew
> > > 
> > > On Mon, Oct 10, 2011 at 7:33 PM, Vincent Toups
> > > reply@reply.github.com
> > > wrote:
> > > 
> > > > Matthew,
> > > > 
> > > > I've pushed a new version of the repository to github after deleting
> > > > the offending file.
> > > > 
> > > > Incidentally, are you on windows?  I'm able to clone the repository
> > > > without any problems using command line get in Ubuntu Natty.
> > > > 
> > > > I've never extensively tested these utilities on Windows, but I think
> > > > only the scripting.el code would be effected.
> > > > 
> > > > -V
> > > > 
> > > > On Mon, Oct 10, 2011 at 1:26 PM, J. Vincent Toups

**Comment by mdaniel** [comment]
Silly me, I should have just checked using the source browser but
instead I waited until I got back to work (and to the Windows
machine).

I am still showing the file in "master":

https://github.com/VincentToups/emacs-utils/blob/master/ls:%20cannot%20access%20page-*.el:%20No%20such%20file%20or%20directory

**Comment by VincentToups** [comment]
It is my fault, anyway.  I deleted the file but forget to "git rm" it.
 I've uploaded the new version.

Incidentally, if you are interested in monadic parser combinators,
there is now a "better-monad-parse.el" which implements them in a much
cleaner, easier to understand and regular fashion.  Policy is
inconsistent in monad-parse.el, but in better-monad-parse, all parsers
start with "=", ie "=item" is the parser which parses one thing from
the input, and all functions which produce parsers start with "=>", as
in "=>or," which produces a parser which succeeds if either of its
arguments succeeds.

In addition to that, better-monad-parse handles parsing of lists,
strings and buffers much more intelligently than monad-parse.  EIEIO
doesn't provide flexible enough dispatch, so I replaced it with my own
implementation of multimethods.  The new library is a lot cleaner, and
it even has unit tests (better-monad-parse-test.el, which also serves
as example usage.)

Let me know if you still can't clone the repo.

-V

On Tue, Oct 11, 2011 at 4:40 AM, Matthew Daniel
reply@reply.github.com
wrote:
